crypto/rand/rand_unix.c

```diff
@@ -325,6 +329,90 @@ static struct random_device {
 } random_devices[OSSL_NELEM(random_device_paths)];
 static int keep_random_devices_open = 1;
 
+#   if defined(DEVRANDOM_WAIT)
+static void *shm_addr;
+
+#    if !defined(FIPS_MODE)
+static void cleanup_shm(void)
+{
+    shmdt(shm_addr);
+}
+#    endif
+#   endif
+
+/*
+ * Ensure that the system randomness source has been adequately seeded.
+ * This is done by having the first start of libcrypto, read one bytes from
```

Comment:
maybe add "or wait using select for readability of /dev/random"

Reply:
I've added something a little different which should cover both cases.

An interesting thought: select(2) on /dev/random returns that it is readable which doesn't mean it is still readable when you try to read it.

Reply:
I don't recall exactly, but those things happen unfortunaltely.
My linux patches on that topic were all turned down unfortunately.
The attitude was we wont fix /dev/urandom, use getrandom.

Reply:
Yeah, I'm aware of that attitude.  Distros should stop including /dev/urandom and include something that blocks and waits instead -- a patch that introduced a new minor device number for "safe random" might have a chance and distros could use this instead of the urandom minor device number.  Voting with your feet can work....